tests/server/server-getTags-002.phpt

```diff
@@ -9,21 +9,23 @@ MongoDB\Driver\Server::getTags() with replica set
 require_once __DIR__ . "/../utils/basic.inc";
 
 $manager = new MongoDB\Driver\Manager(URI);
+$command = new MongoDB\Driver\Command(['ping' => 1]);
+$manager->executeCommand(DATABASE_NAME, $command);
 
-$tags = $manager->selectServer(new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY))->getTags();
-echo "dc: ", array_key_exists('dc', $tags) ? $tags['dc'] : 'not set', "\n";
-echo "ordinal: ", array_key_exists('ordinal', $tags) ? $tags['ordinal'] : 'not set', "\n";
-
-$tags = $manager->selectServer(new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY))->getTags();
-echo "dc: ", array_key_exists('dc', $tags) ? $tags['dc'] : 'not set', "\n";
-echo "ordinal: ", array_key_exists('ordinal', $tags) ? $tags['ordinal'] : 'not set', "\n";
+foreach ($manager->getServers() as $server) {
+    $tags = $server->getTags();
+    echo "dc: ", array_key_exists('dc', $tags) ? $tags['dc'] : 'not set', "\n";
+    echo "ordinal: ", array_key_exists('ordinal', $tags) ? $tags['ordinal'] : 'not set', "\n";
+}
```

Comment:
Is this still a potentially flaky test if the order of servers returned by `getServers()` is non-deterministic? If so, perhaps this needs to be rewritten to instead have a function that asserts any server has certain tags. For example:

```php
function assertSomeServerHasTags($servers, $expectedTags) {
    foreach ($servers as $server) {
        if (array_intersect_assoc($expectedTags, $server->getTags()) === $expectedTags) {
            printf("Found server with tags: %s\n", json_encode($expectedTags));
            return;
        }
    }
    printf("No server has tags: %s\n", json_encode($expectedTags));
}

$servers = $manager->getServers();
assertSomeServerHasTags($servers, ['dc' => 'ny', 'ordinal' => 'one']);
assertSomeServerHasTags($servers, ['dc' => 'pa', 'ordinal' => 'two']);
assertSomeServerHasTags($servers, []);
```

My understanding of [`array_intersect_assoc()`](https://www.php.net/manual/en/function.array-intersect-assoc.php) is that its return value uses the same key ordering as the first argument, which makes it possible to do the `===` check above.

Reply:
Good point. I've kept the loop but removed the use of `array_intersect_assoc`, as the last check will match for any server:
```
php > var_dump(array_intersect_assoc([], ['dc' => 'pa', 'ordinal' => 'two']));
array(0) {
}
```

A loose equality check works around this, and since we're expecting the tags in the order that we create them I think that should be safe but still protect us against non-deterministic order of the server returned from `getServers()`.